x-pack/plugins/ai_infra/product_doc_base/server/services/package_installer/steps/create_index.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Logger } from '@kbn/logging';
+import type { ElasticsearchClient } from '@kbn/core/server';
+import type { MappingTypeMapping, MappingProperty } from '@elastic/elasticsearch/lib/api/types';
+import { internalElserInferenceId } from '../../../../common/consts';
+
+export const createIndex = async ({
+  esClient,
+  indexName,
+  mappings,
+  log,
+}: {
+  esClient: ElasticsearchClient;
+  indexName: string;
+  mappings: MappingTypeMapping;
+  log: Logger;
+}) => {
+  log.debug(`Creating index ${indexName}`);
+
+  overrideInferenceId(mappings, internalElserInferenceId);
+
+  await esClient.indices.create({
```

Comment:
Should this be in a try/catch in case the index already exists due to a race condition or similar?

Reply:
In case of race condition, or any other situation where the index is already present, I think we want to fail, as otherwise we would be indexing the documents twice in the index. wdyt?

Reply:
Yeah that's a good point. Let's leave this alone for now.